File: pending/src/main/java/io/mendirl/demo/pending/entity/PendingEntity.java
package io.mendirl.demo.pending.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Builder.Default;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

import static jakarta.persistence.CascadeType.PERSIST;
import static jakarta.persistence.FetchType.EAGER;
import static jakarta.persistence.GenerationType.IDENTITY;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class PendingEntity {

    @Id
    @GeneratedValue(strategy = IDENTITY)
    private Long id;
    @Version
    private Short version;
    private String name;
//    @Default
//    @OneToMany(mappedBy = "pending", cascade = {PERSIST}, fetch = EAGER)
//    private List<TransactionEntity> transactions = new ArrayList<>();

//    public void addTransaction(TransactionEntity transaction) {
//        transactions.add(transaction);
//        transaction.setPending(this);
//    }

}

File: kpending/src/main/kotlin/io/mendirl/demo/pending/entity/PendingEntity.kt
package io.mendirl.demo.pending.entity

import jakarta.persistence.*
import jakarta.persistence.CascadeType.PERSIST
import jakarta.persistence.FetchType.EAGER
import jakarta.persistence.GenerationType.*

@Entity
class PendingEntity(
        var name: String,
        @OneToMany(mappedBy = "pending", cascade = [PERSIST], fetch = EAGER)
        val transactions: MutableList<TransactionEntity> = mutableListOf(),
        @Id
        @GeneratedValue(strategy = IDENTITY)
        val id: Long? = null,
        @Version
        val version: Short? = null
) {
    fun addTransaction(transaction: TransactionEntity) {
        transactions.add(transaction)
        transaction.pending = this
    }
}